### PR TITLE
tests: fuse8: add test for small ~10k keys input

### PR DESCRIPTION
**Pull request by slimsag**
This test case is currently failing. I noticed when implementing the fuse8 filter
in Zig that, currently, the implementation here fails to populate if the number of
keys is below 125k:

```
testing fuse8 with 10k keys
Too many iterations. Are all your keys unique?bug!
```

I'm not sure why this is, and would love tips / help debugging this. cc @jbapple

### CODE REVIEW

File: tests/unit.c [hunk 1]
```diff
@@ -187,8 +187,46 @@ bool testfuse8() {
   return true;
 }
 
+bool testfuse8_small() {
```
Comment (slimsag) [inline tests/unit.c:190]: Note: this is a direct copy of `testfuse8`, only difference is `size` so if you do want to merge this PR later I can refactor this into a function taking a `size` parameter and test a few different sizes.

I just sent this as-is to see if I could get help debugging why this happens

**Comment by slimsag** [comment]
Also FWIW, I tested the xor8/16 and buffered variants with as little as 10 keys, none of them suffer from this issue.

**Comment by jbapple** [comment]
Most filters, including fuse filters, only work with some probability < 1. The smaller `n` is, the more likely failure is in fuse graph construction.

**Comment by lemire** [comment]
As @jbapple hinted, this is almost surely not a "bug" (in the conventional sense). It is a result of the algorithm.

Now, there is an algorithmic fix for these issues and I will be proposing a solution... eventually.

For now, do not use fuse8 for small filters. I know that this sounds harsh, but it is actually a reasonable recommendation.

**Comment by jbapple** [comment]
The construction depends on “peeling” a hypergraph, but that’s not the only way to solve the set of equations. For small enough values, you can throw a SAT solver at it.

**Comment by slimsag** [comment]
Got it, that makes sense, and I _really_ appreciate the help here (I am still trying to read / fully understand the fuse and xor papers, so apologies if this was a naive question!)

Thank you again!

**Comment by lemire** [comment]
Well. The issue is not invalid btw. It is a real issue and your report was not invalid, I was only trying to characterize it as "not a bug" but it is an issue, for sure.

**Comment by slimsag** [comment]
understood :) was just about to open this for tracking purposes (and in hopes you might post there if you do ever propose a solution to that): #21
